Make file fixes to remove duplicate rules and few others

sharidas opened the pull request:
Make file fixes include
1. Remove duplicate rules - build-dep
2. Remove creation of build-dep file after 'make dist'

Added formatted package.json by npm
Updated .gitignore to remove build-dep

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
This PR includes changes:
- [x] Fix the duplicate "build-dep" rule
- [x] Fixed the creation of empty "build-dep" file
- [x] Removed "build-dep" from the .gitignore file
- [x] Removed "build-dep" from the clean rule
- [x] added auto-indented package.json
   -  The reason for checking in this file, is npm was auto-intending the package.json we had. After checking in this change, the package.json is not updated. The npm version used in my local machine is `6.4.1`.

PVince81 commented on Makefile:
```diff
@@ -34,11 +34,8 @@ dist: build-dep js-templates build-src
-build-dep: package.json
-		$(NPM) install --prefix $(NODE_PREFIX) && touch $@
+build-dep: package.json package-lock.json
+		$(NPM) install $(NODE_PREFIX)
```
it's usually a good idea to touch the "node_modules" folder to update its timestamp. this is where the "$@" comes from which means "the name of the rule". It likely comes from a time where the rule was actually called "node_modules" instead of "build-dep"

sharidas commented:
Current state of the PR:
- [x] Replaced the `build-dep` rule with `node_modules` ( as per the review https://github.com/owncloud/impersonate/pull/130#discussion_r238718059)

Kindly review this PR again.

PVince81 commented on Makefile:
```diff
@@ -29,16 +29,14 @@ endif
 
-build-dep: package.json
-		$(NPM) install --prefix $(NODE_PREFIX) && touch $@
+node_modules: package.json package-lock.json
```
add an alias:
`build-dep: node_modules`

this way whoever intended "build-dep" to be a standard rule would still see it working as expeced

sharidas commented:
Updated the PR, and the current state of the PR has the following change:
- [x] Added `build-dep` back to action. 

Review required for this change.

Note: Before making change mentioned https://github.com/owncloud/impersonate/pull/130#issuecomment-444174107. I wrote this change in my previous version, then later I removed it. Because I missed the change `make build-dep` :( Thanks for pointing it out.
